pkg/testutils/toolkit.go

```diff
@@ -0,0 +1,202 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package testutils
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"sort"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	gomegaTypes "github.com/onsi/gomega/types"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Toolkit struct {
+	client.Client
+	EventuallyPoll        string
+	EventuallyTimeout     string
+	ConsistentlyPoll      string
+	ConsistentallyTimeout string
+	BackgroundCtx         context.Context //nolint:containedctx // this is for convenience
+}
+
+// NewToolkit returns a toolkit using the given Client, with some basic defaults.
+// This is the preferred way to get a Toolkit instance, to avoid unset fields.
+func NewToolkit(client client.Client) Toolkit {
+	return Toolkit{
+		Client:                client,
+		EventuallyPoll:        "100ms",
+		EventuallyTimeout:     "1s",
+		ConsistentlyPoll:      "100ms",
+		ConsistentallyTimeout: "1s",
+		BackgroundCtx:         context.Background(),
+	}
+}
+
+// cleanlyCreate creates the given object, and registers a callback to delete the object which
+// Ginkgo will call at the appropriate time. The error from the `Create` call is returned (so it
+// can be checked) and the `Delete` callback handles 'NotFound' errors as a success.
+func (tk Toolkit) CleanlyCreate(ctx context.Context, obj client.Object, opts ...client.CreateOption) error {
+	createErr := tk.Create(ctx, obj)
```

Comment:
This is generally good advice, but the expectation is that users will always check the return themselves, like `Expect(CleanlyCreate(obj)).To(Succeed())`. And in that case, if there is an error, it will be logged.